libbeat/kibana/index.go

```diff
@@ -0,0 +1,121 @@
+package kibana
+
+import (
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+
+	"github.com/elastic/beats/libbeat/common"
+)
+
+type Index struct {
+	Version   string
+	IndexName string
+	BeatDir   string
+	BeatName  string
+
+	fieldsYaml       string
+	targetDirDefault string
+	targetDir5x      string
+	targetFilename   string
+}
+
+// Create the Index-Pattern for Kibana for 5.x and default.
+func (i *Index) Create() ([]string, error) {
+	indices := []string{}
+
+	err := i.init()
```

Comment:
The pattern we often follow in Beats is to have `NewIndex` function which takes 4 params in this case and returns an `Index` instance. Then `Create` would be run on the object as you have here. So this would probably move the `init` part to the `NewIndex` function.

But that is refactoring we can still do later.

Reply:
thanks for pointing that out, I will change it in the follow up PR then.